src/client-provider.ts

```diff
@@ -305,30 +306,38 @@ export class ZkBobProvider {
         return amountWei / denominator;
     }
 
+    // Round up the fee if needed with fixed fee decimal places (after point)
+    protected async roundFee(fee: bigint): Promise<bigint> {
+        const feeDecimals = this.pool().feeDecimals;
+        if (feeDecimals !== undefined) {
+            const denomLog = (await this.denominator()).toString().length - 1;
+            const poolResDigits = 18 - denomLog;
+            if (poolResDigits > feeDecimals) {
+                const rounder = 10n ** BigInt(poolResDigits - feeDecimals);
+                return (fee / rounder) * rounder + (fee % rounder > 0n ? rounder : 0n);
```

Comment:
Will just leave it here, not sure if it looks simpler though
```suggestion
                return fee % rounder > 0n ? fee + rounder - fee % rounder : fee;
```

Reply:
Is it more efficient and/or clear?

Reply:
Tried to come up with something a bit cleaner/shorter. There is no any difference in efficiency, so feel free to choose any option.

Reply:
Okay, I'll select your case due to the length criteria :)